jira/mapping.go

```diff
@@ -0,0 +1,104 @@
+package jira
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// mapFieldsToStruct maps JIRA fields to a target struct using the field mappings.
+func mapFieldsToStruct[T any](issue *JiraIssue, config *JiraConfig) (T, error) {
+	var result T
+
+	if issue == nil {
+		return result, errors.New("nil issue")
+	}
+	if config == nil {
+		return result, errors.New("nil config")
+	}
+
+	remappedData := make(map[string]interface{}, len(config.FieldMaps))
```

Comment:
I think usually the linter flags this as:


```suggestion
	remappedData := make(map[string]any, len(config.FieldMaps))
```